### PR TITLE
build: generate nightly and release changelogs automatically

### LINKED ISSUE #19645: Feature Request: Automate Changelogging
### Background
Currently, our process for change logging is to create a PR, consider if the change belongs in changelog, add an entry, then force push the whole thing, referencing the PR.

Until late September, we didn't have a changelog in ui at all. It's an important concept for product and product marketing, but it's something we've let slide as engineers.

### Considerations
For a changelog to reference the PR it's part of, the PR has to be created before it can be linked in the changelog.

The relevant entries to include are `feat` and `fix`.

Should this be fully automated, or a script someone has to run?

### PR DESCRIPTION
**Pull request by codyshepherd**
Closes #19645 for `2.0`

* Changelogs are now handled by automation on a nightly and release basis.
* Changelogs will now be uploaded to S3 instead of committed to VCS.
* For nightly builds, changelogs will be put in dl.influxdata.com/platform/nightlies/2.0, and for releases, in dl.influxdata.com/influxdb/releases.
* Release changelogs will be suffixed with the release tag.

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [ ] Tests pass

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -109,14 +109,16 @@ workflows:
...
 
   nightly:
-    triggers:
-      - schedule:
-          cron: "0 5 * * *"
-          filters:
-            branches:
-              only:
-                - "2.0"
+    #triggers:
```
Comment (wbaker85) [inline .circleci/config.yml:112]: Reminder to change this testing code back prior to merging

File: .circleci/config.yml [hunk 2]
```diff
@@ -902,3 +909,70 @@ jobs:
...
+            NEWEST_COMMIT=${CIRCLE_SHA1}
+            TIMESTAMP="$(date -u '+%Y%m%d')"
+            TAG=${TIMESTAMP}
+            if [ -n "${CIRCLE_TAG}" ]; then
+              TAG=${CIRCLE_TAG}
+            fi
+
+            ./scripts/ci/update-changelog.sh \
+              --commit-range "$LAST_COMMIT..$NEWEST_COMMIT" \
+              --prepend ${CHANGELOG_FILENAME} \
```
Comment (wbaker85) [inline .circleci/config.yml:972]: This release changelog should always be a new file right? If so, would it make sense to not use the `--prepend` flag for release changelogs?

File: scripts/ci/update-changelog.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  if ! command -v $dependency &>/dev/null
+  then
+    echo "error: $dependency was not found in PATH" >&2
+    exit 255
+  fi
+done
+
+# The default "starting" commit is a somewhat arbitrary starting point for
+# cataloging recent commits in a way that breaks from the old convention
+DEFAULT_START_COMMIT="891e7d47827e99947e46c82f509e479aa13acf24"
```
Comment (wbaker85) [inline scripts/ci/update-changelog.sh:17]: Should this be `e91d41810f3f44061db544e771f0fc14515aff84`? Or something else on the `2.0` branch?

File: .circleci/config.yml (hunk 2 above)
Comment (codyshepherd) [reply .circleci/config.yml:972]: oh yeah, I suppose that's true. Good catch.

File: scripts/ci/update-changelog.sh (hunk 3 above)
Comment (codyshepherd) [reply scripts/ci/update-changelog.sh:17]: oh hmm, I thought I changed it to the 2.0.8 commit. i'll fix

**Review by wbaker85 — APPROVED** [review]
LGTM!

File: .circleci/config.yml (hunk 1 above)
Comment (codyshepherd) [reply .circleci/config.yml]: :white_check_mark:

**Pull request merged**